go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/distribution/distribution/v3
 
-go 1.22.0
+go 1.21.0
```

Comment:
Does it still allow the version to be specified without `.0` ? That would also avoid some of this.


More fun facts; it's likely that this also _downgrades_ language version; i.e. even if you have a (say) `_go1.22.go` file to provide compatibility for multiple go versions, Go may now ignore those, and just fail to compile "must use goXXX".  See https://github.com/opencontainers/runc/pull/4277#issuecomment-2114392288

Reply:
>  it's likely that this also downgrades language version

I think that's what we want, right? If we want b/w compatibility with one minor version back we should make sure we don't use the language features that are incompatible with those

Reply:
Yeah, so the weird thing there was that, it's not even possible to produce support for both through common mechanisms; i.e., talking about constructs like;


`foo_go1.21.go`

```go
//go:build !go1.22

// implementation for go < 1.22
```

`foo_current.go`

```go
//go:build go1.22

// implementation for go >= 1.22
```

Reply:
If we mark `go.mod` with the minimum supported Go version and force the compilation using that very version as this PR attempts to do I am pretty happy with that outcome.